_posts/2021/2021-09-20-compact_blank-remove-empty-strings-from-array-and-hash.md

```diff
@@ -0,0 +1,69 @@
+---
+title: "Use compact_blank to remove empty strings from Arrays and Hashes"
+description: "This is one of those “Nice Things” in Rails"
+layout: article
+category: ruby
+image:
+  base: '2021/compact_blank-remove-empty-strings-from-array-and-hash'
+  alt: "Compact Disc"
+  credit: "Jan Huber"
+  source: "https://unsplash.com/photos/oTCRizM-PUI"
+
+---
+
+Active Support is a library of “useful” things that supports (!) the other parts of Rails. It includes ways to [extend the base Ruby classes](https://guides.rubyonrails.org/active_support_core_extensions.html), such as `Hash`, `String` and `Integer`.
+
+Even the most experienced Rails developer is constantly finding helpful methods in Active Support that are new to them. And useful methods are continually being added in new versions of the library.
+
+We often have to clean up the values of`Hash`es and `Array`s when parsing paramters, or if our code accepts complex input. Ruby provides the `#compact` method to remove `nil` values from an array and (since 2.4) has provided a [the same `#compact` method](https://ruby-doc.org/core-2.4.0/Hash.html#method-i-compact) for Hash.
```

Comment:
*parameters

*has provided the same

Reply:
```suggestion
We often have to clean up the values of`Hash`es and `Array`s when parsing parameters, or if our code accepts complex input. Ruby provides the `#compact` method to remove `nil` values from an array and (since 2.4) has provided [the same `#compact` method](https://ruby-doc.org/core-2.4.0/Hash.html#method-i-compact) for Hash.
```